packages/react-scripts/config/webpack.config.prod.js

```diff
@@ -466,7 +478,21 @@ module.exports = {
     // https://github.com/jmblog/how-to-optimize-momentjs-with-webpack
     // You can remove this if you don't use Moment.js:
     new webpack.IgnorePlugin(/^\.\/locale$/, /moment$/),
-  ],
+    // TypeScript type checking
+    paths.isTypeScript &&
+      new ForkTsCheckerWebpackPlugin({
+        async: false,
```

Comment:
Did you consider to use `async: true`?
We already tried it in the similar project configuration and it speeds a recompilation a lot. From 3s to less then a second. The drawback is that you cannot see a type errors in browser. Another thing is that, as CRA clears the terminal, you will not see a type error when you start a fresh compilation. However, each next recompilation will show you an error correctly.

I'm sure that it is possible to show errors even in the first run. However, I'm not sure how to show them in browser's console.

Reply:
I did try but CRA was never showing type errors after that; maybe there's some way to make CRA show errors async

Reply:
The problem with `async: false` is that it really slows down rebuilds. I have found it better to use `async: true` along with [fork-ts-checker-notifier-webpack-plugin](https://github.com/johnnyreilly/fork-ts-checker-notifier-webpack-plugin)

Reply:
@frankwallis That’s a really good point. For me personally, it could be a good replacement for error messages in browser. However, we still need to show them in terminal.

Reply:
Need help with that, please reply at the main conversation https://github.com/facebook/create-react-app/pull/4837#issuecomment-425283779